staging/src/k8s.io/api/extensions/v1beta1/types.go

```diff
@@ -569,6 +569,19 @@ type IngressList struct {
 
 // IngressSpec describes the Ingress the user wishes to exist.
 type IngressSpec struct {
+	// Class is the name of the IngressClass cluster resource. The associated
```

Comment:
Question for my learning: Why do we add the new fields to older resources here?
Is it for some round-tripping behavior?

Reply:
We do this in networking.v1beta1 as well. :thinking:

Reply:
Yep, it's a strange requirement to enable round-tripping as I understand it. Definitely makes this PR pretty repetitive.

Reply:
Thank you!
Just to make sure, if one goes through the upgrade, and the old and new resource both are available, then magically your old resource (extensions.Ingress) have the new fields as well, correct?

I'm not saying this is a problem but just want to understand this internal detail.